### PR TITLE
[FIX] Improves handling of missing WCv2 Project ID

### PR DESCRIPTION
**Pull request by andreahaku**
Improves handling of missing WCv2 Project ID.

This is to make sure that developers without a WalletConnect v2 Project ID can still work on MetaMask Mobile app in a stable way.

To QA test this we just need to make sure that the app works (a part from WalletConnect related) without problems. This is mainly for developers not having the chance to use a WCv2 project ID that is required for the app to work with WC.

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions,_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

**Screenshots/Recordings**

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/core/WalletConnect/WalletConnectV2.ts [hunk 1]
```diff
@@ -45,6 +45,10 @@ if (Platform.OS === 'android') {
   });
 }
 
+const { PROJECT_ID } = AppConstants.WALLET_CONNECT;
+export const isWC2Enabled =
+  typeof PROJECT_ID === 'string' && PROJECT_ID?.length > 0;
+
```
Comment (digiwand) [inline app/core/WalletConnect/WalletConnectV2.ts:51]: interesting pattern exporting the constant from here. Works for me. it's nice to have it contained. We could also move it to a util based on another pattern I've seen in this codebase? e.g. `core/../util.ts`

**Review by digiwand — DISMISSED** [review]
LGTM!

**Review by sethkfman — DISMISSED** [review]
LGTM

**Comment by shanejonas** [comment]
👍

**Review by tommasini — APPROVED** [review]
LGTM!

**Review by tommasini — APPROVED** [review]
LGTM!

**Comment by christopherferreira9** [comment]
Looking good 🚀

**Pull request merged**